src/field_take_photo.c

```diff
@@ -0,0 +1,832 @@
+#include "field_take_photo.h"
+#include "constants/scrcmd.h"
+#include "field_warp_tasks.h"
+#include "follow_mon.h"
+#include "gf_gfx_loader.h"
+#include "global.h"
+#include "launch_application.h"
+#include "overlay_01_021F1348.h"
+#include "overlay_01_021F72DC.h"
+#include "overlay_01_021F8D80.h"
+#include "overlay_01_021F944C.h"
+#include "overlay_01_021FB4C0.h"
+#include "overlay_01_022053EC.h"
+#include "photo_album.h"
+#include "script_pokemon_util.h"
+#include "task.h"
+#include "unk_02005D10.h"
+#include "unk_0200FA24.h"
+#include "unk_02054E00.h"
+#include "unk_02055244.h"
+#include "unk_020552A4.h"
+#include "unk_02067A80.h"
+#include "graphic/camera_viewfinder.naix"
+
+typedef enum FieldViewPhotoTaskState {
+    VIEW_PHOTO_STATE_INIT,
+    VIEW_PHOTO_STATE_SELECT,
+    VIEW_PHOTO_STATE_LOAD,
+    VIEW_PHOTO_STATE_FADE_IN,
+    VIEW_PHOTO_STATE_WAIT_PLAYER,
+    VIEW_PHOTO_STATE_FADE_OUT,
+    VIEW_PHOTO_STATE_HANDLE_NEXT,
+    VIEW_PHOTO_STATE_QUIT,
+    VIEW_PHOTO_STATE_QUIT2,
+} FieldViewPhotoTaskState;
+
+typedef enum FieldTakePhotoTaskState {
+    TAKE_PHOTO_STATE_DUMMY_INIT,
+    TAKE_PHOTO_STATE_INIT_LEAVE_OVERWORLD,
+    TAKE_PHOTO_STATE_SAVE_PLAYER_STATE,
+    TAKE_PHOTO_STATE_INIT_RESTORE_OVERWORLD,
+    TAKE_PHOTO_STATE_POSITION_CAMERA,
+    TAKE_PHOTO_STATE_05,
+    TAKE_PHOTO_STATE_ADJUST_PLAYER_FACING_VEC,
+    TAKE_PHOTO_STATE_PARTY_PHOTO_PLACE_MON_ITER,
+    TAKE_PHOTO_STATE_PARTY_PHOTO_ADJUST_MON_OBJ_ITER,
+    TAKE_PHOTO_STATE_NPC_PHOTO_PLACE_NPC,
+    TAKE_PHOTO_STATE_NPC_PHOTO_ADJUST_NPC,
+    TAKE_PHOTO_STATE_NPC_PHOTO_PLACE_MON,
+    TAKE_PHOTO_STATE_NPC_PHOTO_ADJUST_MON,
+    TAKE_PHOTO_STATE_13,
+    TAKE_PHOTO_STATE_DRAW_CAMERA_GFX,
+    TAKE_PHOTO_STATE_SHUTTER,
+    TAKE_PHOTO_STATE_UNLOAD,
+    TAKE_PHOTO_STATE_RESTORE_PLAYER,
+    TAKE_PHOTO_STATE_RESUME_OVERWORLD_AFTER,
+    TAKE_PHOTO_STATE_ADJUST_FOLLOW_MON,
+    TAKE_PHOTO_STATE_EXIT,
+} FieldTakePhotoTaskState;
+
+typedef enum FieldTakePhotoShutterState {
+    TAKE_PHOTO_SHUTTER_STATE_INIT,
+    TAKE_PHOTO_SHUTTER_STATE_DELAY_BEFORE,
+    TAKE_PHOTO_SHUTTER_STATE_FADE_OUT,
+    TAKE_PHOTO_SHUTTER_STATE_EXPOSURE,
+    TAKE_PHOTO_SHUTTER_STATE_FADE_IN,
+    TAKE_PHOTO_SHUTTER_STATE_WAIT_FADE_IN,
+    TAKE_PHOTO_SHUTTER_STATE_6,
+    TAKE_PHOTO_SHUTTER_STATE_DELAY_AFTER,
+} FieldTakePhotoShutterState;
+
+typedef struct FieldTakePhoto3 {
```

Comment:
there a reason for 3?

Reply:
or rather, is there a better name for this? I'd rather avoid numbering stuff like 2 or 3

Reply:
unknown as of yet

Reply:
either way I'd prefer an alternative name than adding 3 to the end of it